src/output-flex.cpp

```diff
@@ -1294,7 +1294,7 @@ void output_flex_t::delete_from_table(table_connection_t *table_connection,
     assert(table_connection);
     auto const id = table_connection->table().map_id(type, osm_id);
 
-    if (table_connection->table().has_geom_column()) {
+    if (!m_options.create && table_connection->table().has_geom_column()) {
```

Comment:
You should rather skip this when expire is not enabled, which may also be the case for updates.